lib/models/repository-states/present.js

```diff
@@ -39,20 +39,46 @@ export default class Present extends State {
     this.operationStates = new OperationStates({didUpdate: this.didUpdate.bind(this)});
 
     this.commitMessage = '';
+    this.commitMessageTemplate = null;
+    this.fetchInitialMessage();
 
     if (history) {
       this.discardHistory.updateHistory(history);
     }
   }
 
-  setCommitMessage(message) {
+  setCommitMessage(message, {suppressUpdate} = {suppressUpdate: false}) {
     this.commitMessage = message;
+    if (!suppressUpdate) {
+      this.didUpdate();
+    }
+  }
+
+  setCommitMessageTemplate(template) {
+    this.commitMessageTemplate = template;
+  }
+
+  async fetchInitialMessage() {
+    const mergeMessage = await this.repository.getMergeMessage();
+    const template = await this.getCommitMessageTemplate();
+    if (template) {
+      this.commitMessageTemplate = template;
+    }
+    if (mergeMessage) {
+      this.setCommitMessage(mergeMessage);
+    } else if (template) {
+      this.setCommitMessage(template);
+    }
   }
 
   getCommitMessage() {
     return this.commitMessage;
   }
 
+  getCommitMessageTemplate() {
+    return this.git().getCommitMessageTemplate();
+  }
```

Comment:
It's odd to me that `getCommitMessageTemplate()` is a git operation, but `setCommitMessageTemplate()` is an accessor for an instance variable. Can we clarify the names to be less misleading somehow?

Also: should the result of this git operation be cached?

Fake edit: on reflection, caching this would cause some weird behavior because we don't have a filesystem watcher on the commit message template itself, so we wouldn't know when to invalidate that cache entry.

Reply:
hahaha yup Katrina and I went through that exact conversation re caching.

renamed the method to `fetchCommitMessageTemplate`.  I'm not 100% sure that's better, but perhaps folks won't think it's the inverse operation of `setCommitMessageTemplate`?